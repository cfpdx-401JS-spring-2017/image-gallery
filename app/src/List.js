import React, { Component } from 'react';
import PropTypes from 'prop-types';

class List extends Component {

  static propTypes = {
    data: PropTypes.array.isRequired
  }

  render() {
    const { data } = this.props;

    return (
      <div>
        {data.map((img, i) => (
          <ul key={i}>
            <li>
              {img.title}
            </li>
            <li>
              <a href=''>
                {img.url}
              </a>
            </li>
            <li>
              {img.description}
            </li>
          </ul>
        ))}
      </div>
    );
  }
}

export default List;