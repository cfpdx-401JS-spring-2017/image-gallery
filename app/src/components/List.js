import React, { Component } from 'react';
import PropTypes from 'prop-types';

class List extends Component {

  static propTypes = {
    data: PropTypes.array.isRequired
  }

  render() {
    const { data, onDelete } = this.props;

    return (
      <div>
        {data.map((img) => (
          <ul key={img._id}>
            <li className="list-title">
              {img.title}
            </li>
            <li>
              <a href="">
                {img.url}
              </a>
            </li>
            <li>
              {img.description}
            </li>
            <li>
              <button onClick={() => onDelete(img._id)}
                      className="delete-button"
              >
                Delete ðŸ—‘
              </button>
            </li>
          </ul>
        ))}
      </div>
    );
  }
}

export default List;