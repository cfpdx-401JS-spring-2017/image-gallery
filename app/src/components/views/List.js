import React from 'react';
import PropTypes from 'prop-types';
import ListImage from './ListImage';

List.propTypes = {
  poms: PropTypes.arrayOf(PropTypes.shape({
    name: PropTypes.string.isRequired,
    description: PropTypes.string,
    url: PropTypes.string.isRequired
  }))
};

export default function List({ poms, handleDelete }) {
  return (
    <ul>
      {poms.map((pom, i) => <ListImage
        key={pom._id}
        {...pom}
        onDelete={() => handleDelete(pom._id, i)}
      />)}
    </ul>
  );
}