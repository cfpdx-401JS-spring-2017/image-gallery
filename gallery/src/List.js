import React, { Component } from 'react';

class List extends Component {
  constructor() {
    super();
    this.state ={

    };
  }
  

  render() {
    return (
      <div className="List">
        {this.props.images.map((image, i) =>{
            let {title, description, url} = image;
          return (
            <div key={i} style={{ width: '30vw', margin: '5px 5px'}}>
              <h3>{title}</h3> 
              <p>{description}</p> 
              <p>{url}</p>
            </div>
          );
          
        })}
      </div>  
    );
  }
}

export default List;