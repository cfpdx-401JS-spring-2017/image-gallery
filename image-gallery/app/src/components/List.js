import React from 'react';
import PropTypes from 'prop-types';
import DeleteImage from './DeleteImage'


    export default function List (props) {

        return (
            <div>
                {props.images.map(image => (
                    <ul className="list-view" key={image._id}>
                        <li className="image-caption">{image.title} </li>
                        <li className="image-caption">{image.description} </li>
                        <li className="image-caption">{image.url} </li>
                        <li><DeleteImage removeImage={() => props.removeImage(props.albumId, image._id)}/></li>
                    </ul>))}
            </div>


        )
    }

    List.proptypes = {
images:PropTypes.array.isRequired,
onDelete:PropTypes.func.isRequired

};

