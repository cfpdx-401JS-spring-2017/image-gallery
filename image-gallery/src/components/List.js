import React from 'react';
import PropTypes from 'prop-types';
import DeleteImage from './DeleteImage'


    export default function List (props) {
  
        return (
            <div>
                {props.images.map(image => (
                    <ul className="list-view" key={image.url}>
                        <li className="image-caption">{image.title} </li>
                        <li className="image-caption">{image.description} </li>
                        <li className="image-caption">{image.url} </li>
                        <li><DeleteImage onClick={() => this.props.onDelete(image._id)}/></li>
                    </ul>))}
                    <DeleteImage/>
            </div>


        )
    }

    List.proptypes = {
images:PropTypes.array.isRequired
};

