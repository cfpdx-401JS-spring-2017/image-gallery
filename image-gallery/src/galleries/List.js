import React, { Component } from 'react';



export default class List extends Component {
  
    render() {
        return (
            <div>
                {this.props.images.map(image => (
                    <ul key={image.url}>
                        <li >{image.title} </li>
                        <li>{image.description} </li>
                        <li>{image.url} </li>
                    </ul>))}
            </div>


        )
    }


}