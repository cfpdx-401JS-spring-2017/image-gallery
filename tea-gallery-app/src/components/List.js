import React from 'react';
import PropTypes from 'prop-types';
import ListItem from './ListItem';

List.PropTypes = {
  images: PropTypes.array.required,
  onDeleteImage: PropTypes.func.required
};

export default function List(props) {

  return (
    <div>
      <ListItem
        images={props.images}
        onDeleteImage={props.onDeleteImage}
      />
    </div>
  );
}
