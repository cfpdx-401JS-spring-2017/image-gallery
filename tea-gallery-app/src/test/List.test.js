import React from 'react';
import renderer from 'react-test-renderer';
import List from '../components/List';

const images = [
  {
    title: 'tea leaves in teapot',
    description: 'what nice leaves',
    url: '/../public/img/teainpot.jpg'
  },
  {
    title: 'Tetsubin over charcoal',
    description: 'what a nice tetsubin',
    url: '/../public/img/tetsubin-charcoal.jpg'
  },
  {
    title: 'Teapot display',
    description: 'what a nice teapot display',
    url: '/../public/img/teapot-display.jpg'
  }
];

describe('List', () => {
  it('renders the List component', () => {
    const component = renderer.create(<List images={images}/>);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });

});