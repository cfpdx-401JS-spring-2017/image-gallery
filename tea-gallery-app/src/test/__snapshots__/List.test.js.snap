// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List renders the List component 1`] = `
<div>
  <ul>
    <div
      className="list-item-wrapper"
    >
      <li
        className="image-list-item"
      >
        Title: 
        tea leaves in teapot
      </li>
      <li
        className="image-list-item"
      >
        Description: 
        what nice leaves
      </li>
      <li
        className="image-list-item"
      >
        <a
          href="/../public/img/teainpot.jpg"
        >
          Image
        </a>
      </li>
      <button
        onClick={[Function]}
      >
        Delete
      </button>
    </div>
    <div
      className="list-item-wrapper"
    >
      <li
        className="image-list-item"
      >
        Title: 
        Tetsubin over charcoal
      </li>
      <li
        className="image-list-item"
      >
        Description: 
        what a nice tetsubin
      </li>
      <li
        className="image-list-item"
      >
        <a
          href="/../public/img/tetsubin-charcoal.jpg"
        >
          Image
        </a>
      </li>
      <button
        onClick={[Function]}
      >
        Delete
      </button>
    </div>
    <div
      className="list-item-wrapper"
    >
      <li
        className="image-list-item"
      >
        Title: 
        Teapot display
      </li>
      <li
        className="image-list-item"
      >
        Description: 
        what a nice teapot display
      </li>
      <li
        className="image-list-item"
      >
        <a
          href="/../public/img/teapot-display.jpg"
        >
          Image
        </a>
      </li>
      <button
        onClick={[Function]}
      >
        Delete
      </button>
    </div>
  </ul>
</div>
`;
